### PR TITLE
Assignment4 - Álvaro Alonso Lancho

### PR DESCRIPTION
**Pull request by b190057**

### CODE REVIEW

File: Assignment4/b190057-b190057/task07.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0/"), override=False)
+g.parse(github_storage + "/rdf/example6.rdf", format="xml")
+
+ns = Namespace("http://somewhere#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0/")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
```
Comment (dgarijo) [inline Assignment4/b190057-b190057/task07.py:30]: Missing all exercises in rdflib

File: Assignment4/b190057-b190057/task07.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+q1 = prepareQuery(
+    """
+    SELECT ?subclass WHERE {
+        ?subclass rdfs:subClassOf ns:LivingThing
```
Comment (dgarijo) [inline Assignment4/b190057-b190057/task07.py:34]: You should print also the subclasses of subclasses of LivingThing

File: Assignment4/b190057-b190057/task07.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+q5 = prepareQuery("""
+    SELECT ?entity
+    WHERE {
+        ?entity foaf:knows ?p1.
+        ?entity foaf:knows ?2.
+        FILTER (?p1 != ?2)
+    }
+    GROUP BY ?entity
+    HAVING (COUNT(?p1) > 1)""", initNs={"foaf":FOAF}
```
Comment (dgarijo) [inline Assignment4/b190057-b190057/task07.py:104]: Did you try this query? `?2` does not look correct, as it's not a correct variable name. Also, the group, having and count are not needed here

**Review by dgarijo — CHANGES_REQUESTED** [review]
There are errors and missing exercises

**Comment by SerPablo** [comment]
Delivered, no more PRs allowed

**Pull request merged**